hooks/tk-flame_actions.py

```diff
@@ -378,7 +782,273 @@ def _get_batch_path_from_published_files(self, sg_info):
+        elif start_frame == end_frame:
+            frame_range = start_frame
+        else:
+            # Generates back the frame range, now formatted
+            frame_range = "[{}-{}]".format(
+                start_frame, end_frame
+            )
+
+        return {"path": path.replace(formatting_str, frame_range), "start_frame": start_frame, "end_frame": end_frame}
+
+    @staticmethod
+    def _guess_frame_range(path):
+        """
+        Try to get the sequence's frame range from the path
+
+        :param path: Path of the sequence containing a frame pattern
+        :type path: str
+        :return: Tuple containing the first and the last frame number of the sequence or tuple of None if failure
+        :rtype: ( int, int ) or ( None, None )
+        """
+        folder, file_name = ntpath.split(path)
+        match = re.match(r"(.*)(%\d+d)(.+)", file_name)
+
+        if not match:
+            raise FlameActionsError("Cannot detect frame pattern for '%s'" % path)
+
+        frame_list = []
+
+        # The frame numbers have to match a certain number of digits
+        frame_len = int(match.group(2)[1:-1])
+
+        # Lets retrieve all the files that's in the folder of the file to match
+        files = [f for f in os.listdir(folder) if os.path.isfile(os.path.join(folder, f))]
+
+        for f in files:
+            # It doesn't match our pattern
+            if not f.startswith(match.group(1)) or not f.endswith(match.group(3)):
+                continue
+
+            # Lets isolate the potential frame number from the file name
+            frame = f[len(match.group(1)):-len(match.group(3))]
+
+            # It's not a frame that match our pattern
+            if len(frame) != frame_len and not frame.isdigit():
+                continue
+
+            # Let's add the frame number of the file that match our pattern
+            frame_list.append(frame)
+
+        frame_list.sort()
+
+        if len(frame_list) >= 2:
+            # We have at least 2 frame so let's return first and the last frame number
+            return frame_list[0], frame_list[-1]
+        elif len(frame_list) == 1:
+            # We only have one frame that match so let's return the same frame as begin and end frame
+            return frame_list[0], frame_list[0]
+        else:
+            # Let's return None because nothing match our pattern
+            return None, None
+
+    @staticmethod
+    def _exists(media_path):
+        """
+        Checks if the path exists directly or as a sequence
+
+        :param media_path: Potential media path
+        :type media_path: str
+        :return: Return if the media_path exists
+        :rtype: bool
+        """
+
+        # Check if the path exists
+        if os.path.exists(media_path):
+            return True
+
+        folder, file_name = ntpath.split(media_path)
+
+        # Try to check if the path is sa sequence
+        match = re.match(r"(.+\.)((?:\[\d+-\d+\])|(?:\d+))(\..+)", file_name)
```

Comment:
Would be better to use the MIO::FilePattern code exposed thru boost::python instead.